api/requestParams/AuthRequestParams.ts

```diff
@@ -0,0 +1,164 @@
+import { request } from './index';
+import {IsEmpty, IsNotEmpty, IsDefined, ValidateNested, Validate} from 'class-validator';
+import { UserModel } from '../../models/UserModel';
+import { ValidEmail, ValidAccessCode } from '../../types/ApiParams';
+import { LoginRequest, RegistrationRequest, PasswordResetRequest } from '../validators/AuthControllerRequests';
+
+export class RegistrationRequestParam implements request{
+    @IsEmpty()
+    param?:any;
+
+    @IsNotEmpty()
+    @IsDefined()
+    @ValidateNested()
+    body?:RegistrationRequest;
+
+    @IsEmpty()
+    queryParam?:any;
+
+    @IsEmpty()
+    query?:any;
+
+    @IsEmpty()
+    authenticatedUser?:any;
+    
+    @IsNotEmpty()
+    @IsDefined()
+    requestTrace?:string;
+}
+
+export class LoginRequestParam implements request{
+    @IsEmpty()
+    param?:any;
+
+    @IsNotEmpty()
+    @IsDefined()
+    @ValidateNested()
+    body?:LoginRequest;
+
+    @IsEmpty()
+    queryParam?:any;
+
+    @IsEmpty()
+    query?:any;
+
+    @IsEmpty()
+    authenticatedUser?:any;
+    
+    @IsNotEmpty()
+    @IsDefined()
+    requestTrace?:string;
+}
+
+export class resendEmailVerificationRequestParam implements request{
```

Comment:
Nit: classes are uppercase, and these aren't validating request params they're validating the entire request.

```suggestion
export class ResendEmailVerificationRequest implements Request {
```

Reply:
These classes should also go in `/api/validators`.

Reply:
Also tbh I haven't thought much about validating entire requests but I imagine we'd want something like
```typescript
export class ResendEmailVerificationRequest implements IResendEmailVerificationRequest {}
```
Probably worth discussing further in Discord before we make more changes, since validating entire requests is gonna add significantly more types.

Reply:
For now, can we move these changes in `/api/requestParams` to another branch to keep this PR's scope manageable? This PR can just be the `ValidUuid/Email/etc` validators.